### ISSUE TITLE
rcParams as singleton class and manage display properties in a class

### ISSUE THREAD

**Issue by banesullivan** [pyvista/pyvista#320]
We need to document all the options in the `rcParams`. Since `rcParams` is currently a dictionary, it might make sense to change it to a singleton class with properties so that we could document each attribute. I really like [`matplotlib`'s approach](https://matplotlib.org/3.1.1/api/matplotlib_configuration_api.html#matplotlib.RcParams) where they implement checks for the parameters as well.

Also, after #300, the `use_panel` option in the `rcParams` will be deprecated and the easiest way for us to keep backward compatibility would be to have a catch and a warning for that key which could be implemented in this class.

**Comment by banesullivan** [comment]
Also, I've been thinking that it might be nice to have a class that could keep track of all the display properties when rendering via `add_mesh`. There are a ton of arguments to that method and it might drastically clean up the code if we could pass all those keyword arguments to a class that manages them and different variations of each one.

This class could inherit the new `rcParams` class to gain all the default display values so we don't have to make `if xxx is None` checks all over the place

**Comment by banesullivan** [comment]
Closing as there is now a Theme class
